### ISSUE TITLE
Expecto fails to build after 'paket update'

### ISSUE THREAD

**Issue by haf** [haf/expecto#148]
@adamchester @cloudRoutine 

Doing`paket update` with paket versions between 4.1 and 5.0-alpha.6 all fail with:

```
/usr/local/share/dotnet/sdk/1.0.1/Microsoft.Common.CurrentVersion.targets(2628,7): error MSB4057: The target "CreateManifestResourceNames" does not exist in the project. [/Users/h/dev/haf/expecto/Expecto.netcore/Expecto.netcore.fsproj]
```

Here's the diff generated:

```diff
diff --git a/Expecto.BenchmarkDotNet/Expecto.BenchmarkDotNet.fsproj b/Expecto.BenchmarkDotNet/Expecto.BenchmarkDotNet.fsproj
index 733a1da..8706a3d 100644
--- a/Expecto.BenchmarkDotNet/Expecto.BenchmarkDotNet.fsproj
+++ b/Expecto.BenchmarkDotNet/Expecto.BenchmarkDotNet.fsproj
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8" standalone="yes"?>
+﻿<?xml version="1.0" encoding="utf-8" standalone="yes"?>
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <Name>Expecto.BenchmarkDotNet</Name>
@@ -36,6 +36,12 @@
     <Reference Include="System.Numerics" />
   </ItemGroup>
   <ItemGroup>
+    <Content Include="any\netstandard1.6\FSharp.Core.optdata">
+      <Paket>True</Paket>
+    </Content>
+    <Content Include="any\netstandard1.6\FSharp.Core.sigdata">
+      <Paket>True</Paket>
+    </Content>
     <ProjectReference Include="../Expecto/Expecto.fsproj">
       <Name>Expecto.fsproj</Name>
       <Project>{c0d55728-10a9-4a7a-9df9-d2f21f663ac2}</Project>
@@ -65,7 +71,7 @@
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
         <Reference Include="BenchmarkDotNet">
-          <HintPath>..\packages\BenchmarkDotNet\lib\net45\BenchmarkDotNet.dll</HintPath>
+          <HintPath>..\packages\BenchmarkDotNet\lib\net46\BenchmarkDotNet.dll</HintPath>
           <Private>True</Private>
           <Paket>True</Paket>
         </Reference>
@@ -76,7 +82,7 @@
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
         <Reference Include="BenchmarkDotNet.Core">
-          <HintPath>..\packages\BenchmarkDotNet.Core\lib\net45\BenchmarkDotNet.Core.dll</HintPath>
+          <HintPath>..\packages\BenchmarkDotNet.Core\lib\net46\BenchmarkDotNet.Core.dll</HintPath>
           <Private>True</Private>
           <Paket>True</Paket>
         </Reference>
@@ -87,7 +93,7 @@
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
         <Reference Include="BenchmarkDotNet.Toolchains.Roslyn">
-          <HintPath>..\packages\BenchmarkDotNet.Toolchains.Roslyn\lib\net45\BenchmarkDotNet.Toolchains.Roslyn.dll</HintPath>
+          <HintPath>..\packages\BenchmarkDotNet.Toolchains.Roslyn\lib\net46\BenchmarkDotNet.Toolchains.Roslyn.dll</HintPath>
           <Private>True</Private>
           <Paket>True</Paket>
         </Reference>
@@ -98,7 +104,7 @@
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
         <Reference Include="FSharp.Core">
-          <HintPath>..\packages\FSharp.Core\lib\net40\FSharp.Core.dll</HintPath>
+          <HintPath>..\packages\FSharp.Core\lib\net45\FSharp.Core.dll</HintPath>
           <Private>True</Private>
           <Paket>True</Paket>
         </Reference>
@@ -130,6 +136,28 @@
   <Choose>
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
+        <Reference Include="Microsoft.DotNet.InternalAbstractions">
+          <HintPath>..\packages\Microsoft.DotNet.InternalAbstractions\lib\net451\Microsoft.DotNet.InternalAbstractions.dll</HintPath>
+          <Private>True</Private>
+          <Paket>True</Paket>
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
+  <Choose>
+    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
+      <ItemGroup>
+        <Reference Include="Microsoft.DotNet.PlatformAbstractions">
+          <HintPath>..\packages\Microsoft.DotNet.PlatformAbstractions\lib\net451\Microsoft.DotNet.PlatformAbstractions.dll</HintPath>
+          <Private>True</Private>
+          <Paket>True</Paket>
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
+  <Choose>
+    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
+      <ItemGroup>
         <Reference Include="System.AppContext">
           <HintPath>..\packages\System.AppContext\lib\net46\System.AppContext.dll</HintPath>
           <Private>True</Private>
@@ -379,4 +407,4 @@
       </ItemGroup>
     </When>
   </Choose>
-</Project>
+</Project>
\ No newline at end of file
diff --git a/Expecto.FsCheck/Expecto.FsCheck.fsproj b/Expecto.FsCheck/Expecto.FsCheck.fsproj
index c3ddb4e..16340e8 100644
--- a/Expecto.FsCheck/Expecto.FsCheck.fsproj
+++ b/Expecto.FsCheck/Expecto.FsCheck.fsproj
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
@@ -48,6 +48,12 @@
   </Choose>
   <Import Project="$(FSharpTargetsPath)" Condition="Exists('$(FSharpTargetsPath)')" />
   <ItemGroup>
+    <Content Include="any\netstandard1.6\FSharp.Core.optdata">
+      <Paket>True</Paket>
+    </Content>
+    <Content Include="any\netstandard1.6\FSharp.Core.sigdata">
+      <Paket>True</Paket>
+    </Content>
     <Compile Include="FsCheck.fs" />
     <Content Include="paket.references" />
   </ItemGroup>
@@ -78,11 +84,22 @@
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
         <Reference Include="FSharp.Core">
-          <HintPath>..\packages\FSharp.Core\lib\net40\FSharp.Core.dll</HintPath>
+          <HintPath>..\packages\FSharp.Core\lib\net45\FSharp.Core.dll</HintPath>
           <Private>True</Private>
           <Paket>True</Paket>
         </Reference>
       </ItemGroup>
     </When>
   </Choose>
-</Project>
+  <Choose>
+    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
+      <ItemGroup>
+        <Reference Include="System.ValueTuple">
+          <HintPath>..\packages\System.ValueTuple\lib\netstandard1.0\System.ValueTuple.dll</HintPath>
+          <Private>True</Private>
+          <Paket>True</Paket>
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
+</Project>
\ No newline at end of file
diff --git a/Expecto.Sample/App.config b/Expecto.Sample/App.config
index 3918676..91448d2 100644
--- a/Expecto.Sample/App.config
+++ b/Expecto.Sample/App.config
@@ -7,17 +7,17 @@
   <dependentAssembly>
     <Paket>True</Paket>
     <assemblyIdentity name="FSharp.Core" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
-    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="4.3.1.0" />
+    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="4.4.1.0" />
   </dependentAssembly>
   <dependentAssembly>
     <Paket>True</Paket>
     <assemblyIdentity name="Microsoft.CodeAnalysis" publicKeyToken="31bf3856ad364e35" culture="neutral" />
-    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="2.0.0.0" />
+    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="2.1.0.0" />
   </dependentAssembly>
   <dependentAssembly>
     <Paket>True</Paket>
     <assemblyIdentity name="Microsoft.CodeAnalysis.CSharp" publicKeyToken="31bf3856ad364e35" culture="neutral" />
-    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="2.0.0.0" />
+    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="2.1.0.0" />
   </dependentAssembly>
   <dependentAssembly>
     <Paket>True</Paket>
diff --git a/Expecto.Sample/Expecto.Sample.fsproj b/Expecto.Sample/Expecto.Sample.fsproj
index ac2d948..721e384 100644
--- a/Expecto.Sample/Expecto.Sample.fsproj
+++ b/Expecto.Sample/Expecto.Sample.fsproj
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8" standalone="yes"?>
+﻿<?xml version="1.0" encoding="utf-8" standalone="yes"?>
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <Name>Expecto.Sample</Name>
@@ -37,6 +37,12 @@
     <Reference Include="System.Numerics" />
   </ItemGroup>
   <ItemGroup>
+    <Content Include="any\netstandard1.6\FSharp.Core.optdata">
+      <Paket>True</Paket>
+    </Content>
+    <Content Include="any\netstandard1.6\FSharp.Core.sigdata">
+      <Paket>True</Paket>
+    </Content>
     <ProjectReference Include="../Expecto/Expecto.fsproj">
       <Name>Expecto.fsproj</Name>
       <Project>{c0d55728-10a9-4a7a-9df9-d2f21f663ac2}</Project>
@@ -75,7 +81,7 @@
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
         <Reference Include="BenchmarkDotNet.Core">
-          <HintPath>..\packages\BenchmarkDotNet.Core\lib\net45\BenchmarkDotNet.Core.dll</HintPath>
+          <HintPath>..\packages\BenchmarkDotNet.Core\lib\net46\BenchmarkDotNet.Core.dll</HintPath>
           <Private>True</Private>
           <Paket>True</Paket>
         </Reference>
@@ -86,7 +92,29 @@
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
         <Reference Include="FSharp.Core">
-          <HintPath>..\packages\FSharp.Core\lib\net40\FSharp.Core.dll</HintPath>
+          <HintPath>..\packages\FSharp.Core\lib\net45\FSharp.Core.dll</HintPath>
+          <Private>True</Private>
+          <Paket>True</Paket>
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
+  <Choose>
+    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
+      <ItemGroup>
+        <Reference Include="Microsoft.DotNet.InternalAbstractions">
+          <HintPath>..\packages\Microsoft.DotNet.InternalAbstractions\lib\net451\Microsoft.DotNet.InternalAbstractions.dll</HintPath>
+          <Private>True</Private>
+          <Paket>True</Paket>
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
+  <Choose>
+    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
+      <ItemGroup>
+        <Reference Include="Microsoft.DotNet.PlatformAbstractions">
+          <HintPath>..\packages\Microsoft.DotNet.PlatformAbstractions\lib\net451\Microsoft.DotNet.PlatformAbstractions.dll</HintPath>
           <Private>True</Private>
           <Paket>True</Paket>
         </Reference>
@@ -104,4 +132,15 @@
       </ItemGroup>
     </When>
   </Choose>
-</Project>
+  <Choose>
+    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
+      <ItemGroup>
+        <Reference Include="System.ValueTuple">
+          <HintPath>..\packages\System.ValueTuple\lib\netstandard1.0\System.ValueTuple.dll</HintPath>
+          <Private>True</Private>
+          <Paket>True</Paket>
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
+</Project>
\ No newline at end of file
diff --git a/Expecto.Tests.CSharp/Expecto.Tests.CSharp.csproj b/Expecto.Tests.CSharp/Expecto.Tests.CSharp.csproj
index 1aa2ef7..548e7c6 100644
--- a/Expecto.Tests.CSharp/Expecto.Tests.CSharp.csproj
+++ b/Expecto.Tests.CSharp/Expecto.Tests.CSharp.csproj
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
   <PropertyGroup>
@@ -46,6 +46,12 @@
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
+    <Content Include="any\netstandard1.6\FSharp.Core.optdata">
+      <Paket>True</Paket>
+    </Content>
+    <Content Include="any\netstandard1.6\FSharp.Core.sigdata">
+      <Paket>True</Paket>
+    </Content>
     <Compile Include="Tests.cs" />
   </ItemGroup>
   <ItemGroup>
@@ -70,11 +76,22 @@
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
         <Reference Include="FSharp.Core">
-          <HintPath>..\packages\FSharp.Core\lib\net40\FSharp.Core.dll</HintPath>
+          <HintPath>..\packages\FSharp.Core\lib\net45\FSharp.Core.dll</HintPath>
           <Private>True</Private>
           <Paket>True</Paket>
         </Reference>
       </ItemGroup>
     </When>
   </Choose>
-</Project>
+  <Choose>
+    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
+      <ItemGroup>
+        <Reference Include="System.ValueTuple">
+          <HintPath>..\packages\System.ValueTuple\lib\netstandard1.0\System.ValueTuple.dll</HintPath>
+          <Private>True</Private>
+          <Paket>True</Paket>
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
+</Project>
\ No newline at end of file
diff --git a/Expecto.Tests.CSharp/app.config b/Expecto.Tests.CSharp/app.config
index 0af883b..b56a575 100644
--- a/Expecto.Tests.CSharp/app.config
+++ b/Expecto.Tests.CSharp/app.config
@@ -4,6 +4,6 @@
   <dependentAssembly>
     <Paket>True</Paket>
     <assemblyIdentity name="FSharp.Core" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
-    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="4.3.1.0" />
+    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="4.4.1.0" />
   </dependentAssembly>
 </assemblyBinding></runtime></configuration>
diff --git a/Expecto.Tests/Expecto.Tests.fsproj b/Expecto.Tests/Expecto.Tests.fsproj
index 5426cff..9a47116 100644
--- a/Expecto.Tests/Expecto.Tests.fsproj
+++ b/Expecto.Tests/Expecto.Tests.fsproj
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
@@ -51,6 +51,12 @@
   </Choose>
   <Import Project="$(FSharpTargetsPath)" Condition="Exists('$(FSharpTargetsPath)')" />
   <ItemGroup>
+    <Content Include="any\netstandard1.6\FSharp.Core.optdata">
+      <Paket>True</Paket>
+    </Content>
+    <Content Include="any\netstandard1.6\FSharp.Core.sigdata">
+      <Paket>True</Paket>
+    </Content>
     <Compile Include="Prelude.fs" />
     <Compile Include="Tests.fs" />
     <Compile Include="SummaryTests.fs" />
@@ -88,7 +94,7 @@
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
         <Reference Include="BenchmarkDotNet">
-          <HintPath>..\packages\BenchmarkDotNet\lib\net45\BenchmarkDotNet.dll</HintPath>
+          <HintPath>..\packages\BenchmarkDotNet\lib\net46\BenchmarkDotNet.dll</HintPath>
           <Private>True</Private>
           <Paket>True</Paket>
         </Reference>
@@ -99,7 +105,7 @@
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
         <Reference Include="BenchmarkDotNet.Core">
-          <HintPath>..\packages\BenchmarkDotNet.Core\lib\net45\BenchmarkDotNet.Core.dll</HintPath>
+          <HintPath>..\packages\BenchmarkDotNet.Core\lib\net46\BenchmarkDotNet.Core.dll</HintPath>
           <Private>True</Private>
           <Paket>True</Paket>
         </Reference>
@@ -110,7 +116,7 @@
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
         <Reference Include="BenchmarkDotNet.Toolchains.Roslyn">
-          <HintPath>..\packages\BenchmarkDotNet.Toolchains.Roslyn\lib\net45\BenchmarkDotNet.Toolchains.Roslyn.dll</HintPath>
+          <HintPath>..\packages\BenchmarkDotNet.Toolchains.Roslyn\lib\net46\BenchmarkDotNet.Toolchains.Roslyn.dll</HintPath>
           <Private>True</Private>
           <Paket>True</Paket>
         </Reference>
@@ -132,7 +138,7 @@
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
         <Reference Include="FSharp.Core">
-          <HintPath>..\packages\FSharp.Core\lib\net40\FSharp.Core.dll</HintPath>
+          <HintPath>..\packages\FSharp.Core\lib\net45\FSharp.Core.dll</HintPath>
           <Private>True</Private>
           <Paket>True</Paket>
         </Reference>
@@ -164,6 +170,28 @@
   <Choose>
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
+        <Reference Include="Microsoft.DotNet.InternalAbstractions">
+          <HintPath>..\packages\Microsoft.DotNet.InternalAbstractions\lib\net451\Microsoft.DotNet.InternalAbstractions.dll</HintPath>
+          <Private>True</Private>
+          <Paket>True</Paket>
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
+  <Choose>
+    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
+      <ItemGroup>
+        <Reference Include="Microsoft.DotNet.PlatformAbstractions">
+          <HintPath>..\packages\Microsoft.DotNet.PlatformAbstractions\lib\net451\Microsoft.DotNet.PlatformAbstractions.dll</HintPath>
+          <Private>True</Private>
+          <Paket>True</Paket>
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
+  <Choose>
+    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
+      <ItemGroup>
         <Reference Include="System.AppContext">
           <HintPath>..\packages\System.AppContext\lib\net46\System.AppContext.dll</HintPath>
           <Private>True</Private>
@@ -413,4 +441,4 @@
       </ItemGroup>
     </When>
   </Choose>
-</Project>
+</Project>
\ No newline at end of file
diff --git a/Expecto.Tests/app.config b/Expecto.Tests/app.config
index 3a0f273..03b46bb 100644
--- a/Expecto.Tests/app.config
+++ b/Expecto.Tests/app.config
@@ -5,17 +5,17 @@
   <dependentAssembly>
     <Paket>True</Paket>
     <assemblyIdentity name="FSharp.Core" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
-    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="4.3.1.0" />
+    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="4.4.1.0" />
   </dependentAssembly>
   <dependentAssembly>
     <Paket>True</Paket>
     <assemblyIdentity name="Microsoft.CodeAnalysis" publicKeyToken="31bf3856ad364e35" culture="neutral" />
-    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="2.0.0.0" />
+    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="2.1.0.0" />
   </dependentAssembly>
   <dependentAssembly>
     <Paket>True</Paket>
     <assemblyIdentity name="Microsoft.CodeAnalysis.CSharp" publicKeyToken="31bf3856ad364e35" culture="neutral" />
-    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="2.0.0.0" />
+    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="2.1.0.0" />
   </dependentAssembly>
   <dependentAssembly>
     <Paket>True</Paket>
diff --git a/Expecto.netcore/Expecto.netcore.fsproj b/Expecto.netcore/Expecto.netcore.fsproj
index 9df2b89..8531201 100644
--- a/Expecto.netcore/Expecto.netcore.fsproj
+++ b/Expecto.netcore/Expecto.netcore.fsproj
@@ -20,4 +20,5 @@
     <PackageReference Include="System.Diagnostics.FileVersionInfo" Version="4.3.0" />
   </ItemGroup>
   <!--<Import Project="..\.paket\Paket.Restore.targets" />-->
+  <Import Project="..\.paket\Paket.Restore.targets" />
 </Project>
\ No newline at end of file
diff --git a/Expecto/Expecto.fsproj b/Expecto/Expecto.fsproj
index da1be5a..e6d6f93 100644
--- a/Expecto/Expecto.fsproj
+++ b/Expecto/Expecto.fsproj
@@ -35,6 +35,12 @@
     <Reference Include="System.Numerics" />
   </ItemGroup>
   <ItemGroup>
+    <Content Include="any\netstandard1.6\FSharp.Core.optdata">
+      <Paket>True</Paket>
+    </Content>
+    <Content Include="any\netstandard1.6\FSharp.Core.sigdata">
+      <Paket>True</Paket>
+    </Content>
     <Compile Include="..\paket-files\logary\logary\src\Logary.Facade\Facade.fs">
       <Paket>True</Paket>
       <Link>paket-files/Facade.fs</Link>
@@ -75,7 +81,7 @@
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
       <ItemGroup>
         <Reference Include="FSharp.Core">
-          <HintPath>..\packages\FSharp.Core\lib\net40\FSharp.Core.dll</HintPath>
+          <HintPath>..\packages\FSharp.Core\lib\net45\FSharp.Core.dll</HintPath>
           <Private>True</Private>
           <Paket>True</Paket>
         </Reference>
@@ -108,4 +114,15 @@
       </ItemGroup>
     </When>
   </Choose>
+  <Choose>
+    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
+      <ItemGroup>
+        <Reference Include="System.ValueTuple">
+          <HintPath>..\packages\System.ValueTuple\lib\netstandard1.0\System.ValueTuple.dll</HintPath>
+          <Private>True</Private>
+          <Paket>True</Paket>
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
 </Project>
\ No newline at end of file
diff --git a/paket.lock b/paket.lock
index 62634a5..323cede 100644
--- a/paket.lock
+++ b/paket.lock
@@ -3,20 +3,24 @@ FRAMEWORK: NET461
 NUGET
   remote: https://www.nuget.org/api/v2
     Argu (3.7)
-    BenchmarkDotNet (0.10.3)
-      BenchmarkDotNet.Core (>= 0.10.3)
-      BenchmarkDotNet.Toolchains.Roslyn (>= 0.10.3)
-    BenchmarkDotNet.Core (0.10.3)
+    BenchmarkDotNet (0.10.5)
+      BenchmarkDotNet.Core (>= 0.10.5)
+      BenchmarkDotNet.Toolchains.Roslyn (>= 0.10.5)
+    BenchmarkDotNet.Core (0.10.5)
+      Microsoft.DotNet.InternalAbstractions (>= 1.0)
+      Microsoft.DotNet.PlatformAbstractions (>= 1.1.1)
       System.Threading.Tasks.Extensions (>= 4.3)
-    BenchmarkDotNet.Toolchains.Roslyn (0.10.3)
-      BenchmarkDotNet.Core (>= 0.10.3)
-      Microsoft.CodeAnalysis.CSharp (>= 1.3.2)
-      System.Threading.Tasks (>= 4.0.11)
+      System.ValueTuple (>= 4.3)
+    BenchmarkDotNet.Toolchains.Roslyn (0.10.5)
+      BenchmarkDotNet.Core (>= 0.10.5)
+      Microsoft.CodeAnalysis.CSharp (>= 2.0)
+      System.Threading.Tasks (>= 4.3)
     FsCheck (2.8)
       FSharp.Core (>= 3.1.2.5)
-    FSharp.Core (3.1.2.5)
+    FSharp.Core (4.1.12)
+      System.ValueTuple (>= 4.3)
     Microsoft.CodeAnalysis.Analyzers (1.1)
-    Microsoft.CodeAnalysis.Common (2.0)
+    Microsoft.CodeAnalysis.Common (2.1)
       Microsoft.CodeAnalysis.Analyzers (>= 1.1)
       System.AppContext (>= 4.3)
       System.Collections (>= 4.3)
@@ -56,8 +60,10 @@ NUGET
       System.Xml.XDocument (>= 4.3)
       System.Xml.XmlDocument (>= 4.3)
       System.Xml.XPath.XDocument (>= 4.3)
-    Microsoft.CodeAnalysis.CSharp (2.0)
-      Microsoft.CodeAnalysis.Common (2.0)
+    Microsoft.CodeAnalysis.CSharp (2.1)
+      Microsoft.CodeAnalysis.Common (2.1)
+    Microsoft.DotNet.InternalAbstractions (1.0)
+    Microsoft.DotNet.PlatformAbstractions (1.1.1)
     Mono.Cecil (0.10-beta5)
     NuGet.CommandLine (3.5)
     System.AppContext (4.3)
@@ -113,4 +119,4 @@ GITHUB
 GROUP build
 NUGET
   remote: https://www.nuget.org/api/v2
-    FAKE (4.57.4)
+    FAKE (4.60)
diff --git a/testdata/Expecto.Focused.Tests/Expecto.Focused.Tests.fsproj b/testdata/Expecto.Focused.Tests/Expecto.Focused.Tests.fsproj
index 5bc1d4d..e3c74d5 100644
--- a/testdata/Expecto.Focused.Tests/Expecto.Focused.Tests.fsproj
+++ b/testdata/Expecto.Focused.Tests/Expecto.Focused.Tests.fsproj
@@ -44,17 +44,13 @@
   </Choose>
   <Import Project="$(FSharpTargetsPath)" Condition="Exists('$(FSharpTargetsPath)')" />
   <Import Project="..\..\.paket\paket.targets" />
-  <Choose>
-    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
-      <ItemGroup>
-        <Reference Include="FSharp.Core">
-          <HintPath>..\..\packages\FSharp.Core\lib\net40\FSharp.Core.dll</HintPath>
-          <Paket>True</Paket>
-        </Reference>
-      </ItemGroup>
-    </When>
-  </Choose>
   <ItemGroup>
+    <Content Include="any\netstandard1.6\FSharp.Core.optdata">
+      <Paket>True</Paket>
+    </Content>
+    <Content Include="any\netstandard1.6\FSharp.Core.sigdata">
+      <Paket>True</Paket>
+    </Content>
     <Compile Include="Main.fs" />
     <None Include="paket.references" />
     <Content Include="app.config" />
@@ -69,4 +65,26 @@
       <Private>True</Private>
     </ProjectReference>
   </ItemGroup>
+  <Choose>
+    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
+      <ItemGroup>
+        <Reference Include="FSharp.Core">
+          <HintPath>..\..\packages\FSharp.Core\lib\net45\FSharp.Core.dll</HintPath>
+          <Private>True</Private>
+          <Paket>True</Paket>
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
+  <Choose>
+    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.6.1'">
+      <ItemGroup>
+        <Reference Include="System.ValueTuple">
+          <HintPath>..\..\packages\System.ValueTuple\lib\netstandard1.0\System.ValueTuple.dll</HintPath>
+          <Private>True</Private>
+          <Paket>True</Paket>
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
 </Project>
\ No newline at end of file
diff --git a/testdata/Expecto.Focused.Tests/app.config b/testdata/Expecto.Focused.Tests/app.config
index 0af883b..b56a575 100644
--- a/testdata/Expecto.Focused.Tests/app.config
+++ b/testdata/Expecto.Focused.Tests/app.config
@@ -4,6 +4,6 @@
   <dependentAssembly>
     <Paket>True</Paket>
     <assemblyIdentity name="FSharp.Core" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
-    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="4.3.1.0" />
+    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="4.4.1.0" />
   </dependentAssembly>
 </assemblyBinding></runtime></configuration>
```

Pinging @forki

**Comment by haf** [comment]
I'm branching off `releases/v4.x` as a stable release branch, off of master now.

**Comment by cloudRoutine** [comment]
For future reference for netcore projects if you see this manifest resources not created error, try updating the fsharp.net.sdk to the latest version if possible and make sure restore is being called before build, that's when the manifest resources are created. If restore isn't being run make sure that the path to paket.targets is correct in the fsproj and that the `paket.targets` being used is the latest version. If you have paket on your PATH already you can just delete the `.paket` dir and call `paket init` to get the latest versions of the targets.

But that shouldn't apply here because it's the old style fsproj that isn't building. Is it trying to build using msbuild15? if so you need to make sure it uses msbuild14 instead.

if you were using FAKE you'd do this with
```fsharp
let msbuild14 = ProgramFilesX86</>"MSBuild"</>"14.0"</>"Bin"</>"MSBuild.exe"

if isWindows && fileExists msbuild14 then 
    setEnvironVar "MSBUILD"  msbuild14
```

I don't know what you need to do to make it happen for your build system

**Comment by haf** [comment]
I use `xbuild` on `OS X` and not Windows. Repro is pushed on that platform. Can paket update the SDK and will that fix the issue? Why is it suddenly broken?

**Comment by cloudRoutine** [comment]
did the version of xbuild get upgraded? i'm pretty sure this is not a paket issue, but let me try running this on my machine in a little while

**Comment by haf** [comment]
```
xbuild --version
XBuild Engine Version 14.0
Mono, Version 4.8.1.0
Copyright (C) 2005-2013 Various Mono authors
MSBUILD: error MSBUILD0000: Project file '--version' not found.
```

**Comment by cloudRoutine** [comment]
you may not have noticed this, but you have paket hardcoded in the build.cmd and build.sh to `4.0.0-alpha030`

when I run it on my machine it uses the wrong msbuild 
```
Starting Target: Build (==> Clean)
Building project: C:\Users\jared\Github\Forks\expecto\Expecto.sln
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe  C:\Users\jared\Github\Forks\expecto\Expecto.sln /t:Rebuild /m      /p:RestorePackages="False" /p:Configuration="Release" /logger:Fake.MsBuildLogger+ErrorLogger,"C:\Users\jared\Github\Forks\expecto\packages\build\FAKE\tools\FakeLib.dll"
Microsoft (R) Build Engine version 15.1.1012.6693
Copyright (C) Microsoft Corporation. All rights reserved.
```
I add those lines from before to `build.fsx`

```fsharp
let msbuild14 = ProgramFilesX86</>"MSBuild"</>"14.0"</>"Bin"</>"MSBuild.exe"

if isWindows && fileExists msbuild14 then
    setEnvironVar "MSBUILD"  msbuild14
```
```
       CopyFilesToOutputDirectory:
         Copying file from "obj\Release\Expecto.dll" to "bin\Release\Expecto.dll".
         Expecto -> C:\Users\jared\Github\Forks\expecto\Expecto\bin\Release\Expecto.dll
         Copying file from "obj\Release\Expecto.pdb" to "bin\Release\Expecto.pdb".
     2>Done Building Project "C:\Users\jared\Github\Forks\expecto\Expecto\Expecto.fsproj" (Rebuild target(s)).
```
no manifest resource problem

the build still fails though with
```
       "C:\Users\jared\Github\Forks\expecto\Expecto.sln" (Rebuild target) (1) ->
       "C:\Users\jared\Github\Forks\expecto\Expecto.Tests.CSharp\Expecto.Tests.CSharp.csproj" (Rebuild target) (8) ->
       (CoreCompile target) ->
         Tests.cs(3,7): error CS0246: The type or namespace name 'Expecto' could not be found (are you missing a using d
       irective or an assembly reference?) [C:\Users\jared\Github\Forks\expecto\Expecto.Tests.CSharp\Expecto.Tests.CShar
       p.csproj]
         Tests.cs(4,7): error CS0246: The type or namespace name 'Expecto' could not be found (are you missing a using d
       irective or an assembly reference?) [C:\Users\jared\Github\Forks\expecto\Expecto.Tests.CSharp\Expecto.Tests.CShar
       p.csproj]
         Tests.cs(10,10): error CS0246: The type or namespace name 'TestsAttribute' could not be found (are you missing
       a using directive or an assembly reference?) [C:\Users\jared\Github\Forks\expecto\Expecto.Tests.CSharp\Expecto.Te
       sts.CSharp.csproj]
         Tests.cs(10,10): error CS0246: The type or namespace name 'Tests' could not be found (are you missing a using d
       irective or an assembly reference?) [C:\Users\jared\Github\Forks\expecto\Expecto.Tests.CSharp\Expecto.Tests.CShar
       p.csproj]
         Tests.cs(11,23): error CS0246: The type or namespace name 'Expecto' could not be found (are you missing a using
        directive or an assembly reference?) [C:\Users\jared\Github\Forks\expecto\Expecto.Tests.CSharp\Expecto.Tests.CSh
       arp.csproj]
```
but that's probably due to some other configuration issue

like I said before, paket is not the source of this issue

**Comment by haf** [comment]
I haven't actually; build.cmd and build.sh was PRed (by you, perhaps?) I use `bundle exec rake` to build.

**Comment by cloudRoutine** [comment]
you can try changing 
```
<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
to
```
<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
in Expecto.fsproj

**Comment by haf** [comment]
I doubt it's an xbuild problem... It crashes in the dotnet builder?

**Comment by haf** [comment]
But indeed, that did help. Let me try to build it now.

**Comment by haf** [comment]
Thank you @cloudRoutine

**Comment by haf** [comment]
Not solved;

```

         	Specified references-file '/Users/h/dev/haf/expecto/Expecto.netcore/obj/Expecto.netcore.fsproj.references' does not exist.
     1>/Users/h/dev/haf/expecto/.paket/Paket.Restore.targets(59,5): error MSB3073: The command "/Library/Frameworks/Mono.framework/Commands/mono --runtime=v4.0.30319 "/Users/h/dev/haf/expecto/.paket/paket.exe" fix-nuspec file "obj/Expecto.5.0.0-beta3.nuspec" references-file "/Users/h/dev/haf/expecto/Expecto.netcore/obj/Expecto.netcore.fsproj.references" " exited with code 1. [/Users/h/dev/haf/expecto/Expecto.netcore/Expecto.netcore.fsproj]
     1>Done Building Project "/Users/h/dev/haf/expecto/Expecto.netcore/Expecto.netcore.fsproj" (pack target(s)) -- FAILED.
```

when rake runs `dotnet pack -v n --no-build -c Release -o ../build/clipkg /p:Version=5.0.0-beta3 Expecto.netcore/Expecto.netcore.fsproj` from running `bundle exec rake` to start with.

**Comment by cloudRoutine** [comment]
are your `/obj/` dirs getting cleaned before build? detritus in them will cause all kinds of problems

**Comment by haf** [comment]
No, I just build. Isn't the build tool responsible for its own garbage?

**Comment by haf** [comment]
Back to 
/usr/local/share/dotnet/sdk/1.0.1/Microsoft.Common.CurrentVersion.targets(2628,7): error MSB4057: The target "CreateManifestResourceNames" does not exist in the project. [/Users/h/dev/haf/expecto/Expecto.netcore/Expecto.netcore.fsproj]
    0 Warning(s)
    1 Error(s)

**Comment by haf** [comment]
Need to stop fighting windmills for today. /cc @enricosada

**Comment by cloudRoutine** [comment]
one would think that, but this is just one of many stupid things that happens in the brave new world of netcore.  

You have to run restore after cleaning, like I said before restore is what sets up the manifest so that msbuild will run properly.

Msbuild14 needs to be used on the old style fsproj, msbuild15 (aka dotnet build) needs to be used for the new style netcore fsproj

**Comment by haf** [comment]
Could you PR that into the Rakefile, please?

**Comment by cloudRoutine** [comment]
i don't write ruby

**Comment by enricosada** [comment]
> Need to stop fighting windmills for today. /cc @enricosada

Atm (with .net core sdk 1.0), you need to do at least once the `dotnet restore`, before `dotnet build`
And you are also responsible to do that manually if you change `PackageReference` 
see referenced issue for more info

> Back to
> /usr/local/share/dotnet/sdk/1.0.1/Microsoft.Common.CurrentVersion.targets(2628,7): error MSB4057: The target "CreateManifestResourceNames" does not exist in the project.

That happen if prohect is not restored. i know. i was too short on time for 1.0 and didnt see that. i cannot fix it untile next 1.0 or 2.0 release (it's a file embedded in dotnet cli, cannot update it)

added https://github.com/dotnet/netcorecli-fsc/issues/109 to track that

btw @haf @cloudRoutine feel free to open bug in https://github.com/dotnet/netcorecli-fsc/ (fsharp) or https://github.com/dotnet/cli/ if you see it

**Comment by haf** [comment]
@cloudRoutine I've removed you as collaborator from the repository since you don't write ruby and thus cannot keep the master branch compilable.

**Comment by haf** [comment]
I've removed FAKE from the master branch too, just to make it clear that I need to be able to release my project.

**Comment by cloudRoutine** [comment]
don't ask me for help next time

**Comment by haf** [comment]
You can count on it.

**Comment by forki** [comment]
@haf people tried to help you here. can we please be nice to each other?

**Comment by enricosada** [comment]
Yes, please let's just calm a bit.

@haf what you run from command line? `bundle exec rake merge_nupkgs` is enough to check the error?
I just need to check without doing a real release/push to nuget.org.

I see a `dotnet restore` in the `Rakefile`, so strange that doesnt run.
Let me check.

**Comment by haf** [comment]
@forki I'm frustrated that I've now spent close to two hours on this without a resolution. It's not Jared's fault, but it's surely not mine either since I didn't change this piece of infrastructure. I have infrastructure that is streamlined towards releasing code frequently; albacore and the build needs to work from the command line; which is what albacore delegates to. So when the reply is "I don't write ruby", it's the straw that breaks this camel's back.

**Comment by cloudRoutine** [comment]
I don't understand what your problem is, I stop what I'm working on to help you diagnose your problem and I give you detailed advice on how to address them and similar ones that may come up in the future. Even if I did know ruby I wouldn't update the Rakefile right now because I'm busy, but I don't and I have zero familiarity with Rake so it wouldn't make any sense for me to do it.

It's a shame you're so unappreciative and decided to take out your frustration on me.

**Comment by forki** [comment]
@haf everyone in the ecosystem is frustrated about netcore and MSBuild 15. It's not just you.

**Comment by haf** [comment]
**** it you're both right. I added Jared back.

**Comment by enricosada** [comment]
> @haf everyone in the ecosystem is frustrated about netcore and MSBuild 15. It's not just you.

As a note about that (not that i care a lot :D), now also sdk is 1.0 RTM.

So can you please give some structured feedback about it, not just generic rants (not speaking about your comment @forki), in the right repositories?

It's also possible to contribute to that, instead of waiting MS, **at least with bug reports please**. 
Or feedback on flow. or roadblock. or part not nice for existing libraries (that's why for example i added `mergenupkg`, because ppl wait for VS to migrate obv).

I continue to say that, but is not about .net core, is about new fsproj (who is gonna replace old), so if we can just help there (just with bug reports at least) is better for everyone.

Or we can continue to focus only on old fsproj and old sdk, and when we get bugs, that's it. we can just complain.

Btw @haf , cloning locally `master` now, let me see

**Comment by forki** [comment]
@enricosada you are absolutely right. Every single issue we see is fixable. And I appreciate your work to get us there. The problem in my comment was more that stuff is really really flaky and we basically encounter new issues with every update. This is frustrating. I seriously hope this will go away soon but since MSBuild 15 is not backwards compatible I seriously doubt it

**Comment by haf** [comment]
And then there are cascading issues from the way of the main build team;

 - https://github.com/Albacore/albacore/issues/229 – breaking change for no apparent reason
 - https://github.com/Microsoft/msbuild/issues/2003 – response: just ignore that it doesn't work by not using the feature you started using msbuild for
 - https://travis-ci.org/haf/expecto/jobs/226747060#L1861 – this issue
 - https://github.com/SuaveIO/suave/issues/605 – by dotnet core not supporting passing versions on the command line
 - https://ci.appveyor.com/project/haf/suave/build/2.0.633/job/p2f18wsiysbyyask#L754 – when it's building itself (how can it not find itself?)

That's just in the last week. Every week is like this. Almost every day I restart a build due to flaky MSFT servers, broken mono package repos, bugs lingering unfixed in mono for years like HTTP endgetresponse.

**Comment by enricosada** [comment]
> The problem in my comment was more that stuff is really really flaky and we basically encounter new issues with every update

Every update **of a beta/rc/preview version**. You got only 1 rtm.
Not to be pedantic, but you have choosen you go bleeding edge, and use a prerelease version to use .NET Core Sdk.
And **thank you a lot for that**, i fixed lot of bugs in the wild i didnt see in the previous version, and now 1.0 rtm is ok for f# (can be better obv, filling bugs and contributing to it)
**I think was the right thing to do, not for everyone obv, but for the long term**.

Ihmo the most of pain is because we have TWO version in parallel, old fsproj and new fsproj (that's because nobody want to migrate if VS doesnt work). otherwise is a bit easier, like c# projects do, just using new sdk (the new csproj), see Netonsoft.json for example.

> since MSBuild 15 is not backwards compatible

msbuild it is, you can run old fsproj (if targets where not badly written) or you make assumptions.
But there is a new sdk in town, where things are moving for a reason. converge mono/netcore/net in a single sdk and msbuild. to fix quirks, finally. 
fix old sdk (old fsproj/csproj) for mono/vs is not going to happen, new implementation fix it and replace old giving us finally a single stuff who works exactly the same, same command line too.
That mean a new implementation of the target files for build/restore/publish, and others like pack/test etc. That's based on msbuild, but is like say is using the same F# language, but implementation is different.

The old good part, now xbuld is dead with new mono, and msbuild will be always the same, and will work exactly the same **with the new sdk/fsproj, not the old**

But mono5 IS in alpha, so choose your bleeding edge level.

> by dotnet core not supporting passing versions on the command line

Pass `/p:Version=1.2.3-beta.1` or set env var `Version` or set the property `<Version>` to create a nuget Package with a specific version.

Thas env/prop/cmdline was working also on old msbuild, because that's how properties works in msbuild.

> That's just in the last week. Every week is like this. Almost every day I restart a build due to flaky MSFT servers, broken mono package repos, bugs lingering unfixed in mono for years like HTTP endgetresponse

flaky MSFT servers -> ??
mono bugs -> mono bugs, ok, hopefully the code sharing with .net fw will help.

**Comment by enricosada** [comment]
@haf how can improve it? We need an example repo with fake and one with rake? to show how the current status to use it in an existing library? so can be used as reference?
Really asking, because we can do that, if can help.
dunno, other ideas?

**Comment by haf** [comment]
> flaky MSFT servers -> ??

 - The dotnet install script stopped working when they changed where the sdk/cli was hosted a while back
 - That also caused the rakefile to crash because it had been downloading the cli
 - Two days ago NuGet returned 500 Internal Error for half a day when I was restoring packages in the suave builds
 - The mono deb package servers are sometimes updated without proper signatures causing all installs to fail, which also affects releases in production for us
 - The mono deb package servers have the wrong debian name causing excessive warnings on installation
 - NuGet is eventually consistent which causes needless delays when I release the 20-something packages of Logary and bump other libraries in cascade; time I can just spend waiting. Sometimes it takes more than five minutes for a package to be indexed.
 - ... and more; this is just infrastructure.

About bleeding edge level; **I really want stable things**. I've tried to keep down the F# version and tried not to change any build scripts needlessly just to avoid instability. (But I get pushback in the form of "why do your libraries always target such a low F# version?") I run the mono that's most stable; 4.4.0 had a broken garbage collector so I had to move up; 4.6.0 halved performance on my workload and 4.8.0, well, it had fixed a few bugs I needed fixing AND finally 5 years later supported TLS 1.2 that I need. I'm not at all using v5 of Mono and I hope I can stick on 4.8 for a few more years if I can just get a stable environment; but that version keeps failing with either null ref exception or ObjectDisposedException in EndGetRequest, which is a problem for me, running a web workload. So you tell me what I should do?

Like I've stated before, I appreciate the work you've been doing.

Anyway; to answer your question. What I'd like is a command-line only, for linux and osx for setting up a F# project. It doesn't have to use a "build system" like Rake or FAKE, it's enough to use the command line for installing the prerequisites and build and to have a .Net 4.6.1 (latest mono supported .Net version) and .Net Core vSomething + merging + publishing on command line + a description of what and when things will change. That would be great. Part of the equation is that I have absolutely NO news channel on what's going on with .Net Core besides you. People port libraries sometimes, but mostly people complain when I don't do it for them; but similarly the tooling seems broken which this issue and the linked issues demonstrate. It's draining.

It would be good if the .Net Core would work with the F# language as part of the core and both their and your communication comes in the form of public minutes from meetings over Google Hangouts, like Kubernetes is doing it (to name a project that has good leadership).

**Comment by enricosada** [comment]
> About bleeding edge level; I really want stable things.

Me too, but netcore was completly new, so i'd like to see bugs to stabilize asap.
mono is another story.

> The dotnet install script stopped working when they changed where the sdk/cli was hosted a while back

@haf that was my same pain ( I know brew doesnt work yet, is not a small task but i understand why is not yet done ).

But the install script is a good example for everyone in this discussion.
I asked `dotnet/cli` team about it in https://github.com/dotnet/cli/issues/5952 

Now instead of 

```
https://github.com/dotnet/cli/blob/rel/1.0.0/scripts/obtain/dotnet-install.ps1
```

there is a stable

```
https://dot.net/v1/dotnet-install.sh
```

and there is [documentation about that](https://docs.microsoft.com/en-us/dotnet/articles/core/tools/dotnet-install-script)

Small improvements? yes, but from a single bug report.

> That would be great. Part of the equation is that I have absolutely NO news channel on what's going on with .Net Core besides you

That's an issue. I'd like more feedback how to address that. Maybe i am not the best person, because i code more than talk (this issue excluded).
That's also why i am writing there, because maybe i am too much in the loop and i dont see that issue.
So i sue this as opportunity to gather feedback.

@forki we really really really need to address this as part of `The F# Core Engineering Group` ihmo /cc @dsyme @7sharp9 @tpetricek because now is not a beta anymore, and maybe we are too much on slack with the same ppl (and lots are skilled in F# but doesnt know the the status of tooling atm or where is going, not just .net core, but the sdk or rest of todo). 
Current situation is more a wip than it need to be, because i see milestones (1.0 sdk, etc), but ppl see just  a continuous flow of changes, and is wrong, is not needed.

Maybe we just need to list for F# what's stable, what's not, and update notes.

I have https://github.com/dotnet/netcorecli-fsc/wiki/Status as status for F# and new sdk/.net core

> both their and your communication comes in the form of public minutes from meetings

My minutes are release notes (new work) and the wiki. meeting is just with myself :(
.net team and cli already does that, but is a really large surface (.net core, sdk, msbuild, VS), so need a bit of knowledge, **otherwise ppl think `Core = new Sdk` and that's wrong**, and need some background, but after than roadmap is pretty clear.

> Anyway; to answer your question. What I'd like is a command-line only, for linux and osx for setting up a F# project.

That's what the new sdk is going to be, or a possibility at least. 
And will work the same on mono/windows/osx. for all target frameworks (net45, netstandard, netcoreapp). With `dotnet new` for scaffolding, and `dotnet add` to modify the fsprj/csproj

So a single fsproj contains multiple target frameworks, and you can `dotnet build` and `dotnet pack`. That's it. or `dotnet publish` for docker, or anything else is needed (`dotnet test`)

Will work (already) from any of the three msbuild implementation:

- mono: `msbuild /t:Pack`
- win: `msbuild /t:Pack`
- core: `dotnet msbuild /t:Pack` (that's what `dotnet pack` runs, because dotnet cli is nicer to use)

Two days ago (https://github.com/dotnet/netcorecli-fsc/releases/tag/v1.0.3) i pushed the working initial work for F# (mono is removing a workaround already).

That's really really nice ihmo. I'd like to see that asap out of bleeding edge.

**Comment by haf** [comment]
This information is really valuable. Would it be possible to ask you to jot down some of this in a blog entry every once in a while?
